### ISSUE TITLE
Feature request: generic workers from script root.

### ISSUE THREAD

**Issue by peterhuene** [Azure/azure-functions-host#3139]
This is a feature request relating to [Azure Functions language extensibility](https://github.com/Azure/azure-functions-host/wiki/Language-Extensibility) and not a product issue.

The generic worker configuration currently makes the following assumptions:

* The worker executable (`DefaultExecutablePath`) will be on the PATH (e.g. `java`, `node`, etc).
* The worker executable accepts an argument to the "worker implementation" (`DefaultWorkerPath`) (e.g. `azure-functions-java-worker.jar`, `nodejsWorker.js`, etc.).

This works great for languages that have a VM or interpreter to run the user's functions.

However, for languages like Rust ([see my experimental implementation here](https://github.com/peterhuene/azure-functions-rs/)) that compile to native executables, I envision the worker executable to be built from the user's source code itself and deployed inside of script root.  Think of it like pre-built C# Azure Functions, except instead of an assembly implementing only the user's functions, what's generated is the worker executable that communicates with the host and invokes the function implementations in-proc.

Long story short: I'd love to see the ability to describe a generic worker that lives inside of script root.  I have a [commit I've made to my fork](https://github.com/peterhuene/azure-functions-host/commit/df0cc4ca18f75c25aaf9edb8a3e5ea54f8335bdb) that attempts to implement just that, but there may be a better way to accomplish it (ignore the change to reference the Rust worker package, though; that's just for my experimentation).

Thank you!

**Comment by peterhuene** [comment]
I'm going to close this for now since I think I have a better approach given the move to using containers for the host anyway:

* Put the worker executable on the PATH in the image.
* Take an additional "command" argument from the worker executable that can be passed via `DefaultWorkerPath` (convoluted, but should work).

**Comment by peterhuene** [comment]
I did manage to get it to work with a vanilla nightly Azure Functions Host:

Source: https://github.com/peterhuene/azure-functions-rs/blob/master/examples/http/src/greet.rs
Deployed: http://rustapp.azurewebsites.net/api/greet
